### ISSUE TITLE
Cache item throw an error on reading with DateTimeImmutable date objects

### ISSUE THREAD

**Issue by jacekkarczmarczyk** [PHPSocialNetwork/phpfastcache#860]
**Configuration**
- **PhpFastCache version:** 9.1
- **PhpFastCache API version:** 4.1.0
- **PHP version:** 8.1
- **Operating system:** win10

**Describe the bug**

When updating cache item like this:

```php
        $item = $cacheItemPool->getItem('key');
        $item->set('some data');
        $item->setExpirationDate(new DateTimeImmutable('+1 year'));
        $cacheItemPool->save($item);
```

(which is valid as `setExpirationDate` allows `DateTimeInterface`)

phpfastcache fails on reading data:

```
TypeError: DateTime::createFromFormat(): Argument #2 ($datetime) must be of type string, DateTimeImmutable given
```

in `\Phpfastcache\Core\Pool\DriverBaseTrait::driverUnwrapEdate`

**Expected behavior**
No error when expiration date set to `DateTimeImmutable`

**Comment by Geolim4** [comment]
Hello,

Nice catch, but I'm facing one potential problem with **DateTimeImmutable** object.

I'm not sure how atomic operations such as `$item::expiresAfter()` should react if Immutable object are set 🤔 

I think that I should convert immutable object to mutable object before passing them to the core.

**Comment by Geolim4** [comment]
After some code review, I've decided that Immutable DateTime will be automatically converted to Mutable DateTime for one simple reason: Very few backend support "plain DateTime" object storage. In fact for most of them I have to store the date as an ATOM formatted string .

So I cannot even persist the mutability status of the Datetime object along with other stored data.

**Comment by jacekkarczmarczyk** [comment]
Every solution is fine as long as it doesn't throw error when reading data saved with DateTimeImmutable expiration date :)

**Comment by Geolim4** [comment]
Indeed, however if I fix that error I will introduce an unintended side-effect that I'll need to fix too, so I need to fix both of them.

**Comment by Geolim4** [comment]
Can you test the v9 branch please ?

If it's ok for you I'll push a fix for v8.1 tomorrow too.

Thanks.

**Comment by jacekkarczmarczyk** [comment]
Thank you for the patch, I have some comments and problems:
- how to test v9 branch? ;) When I did `composer require phpfastcache/phpfastcache:v9` it installed version 9.0, when I did `composer require phpfastcache/phpfastcache:dev-v9` I got an error `Could not find package phpfastcache/phpfastcache in a version matching "dev-v9" and a stability matching "dev".` (I have `"minimum-stability": "dev",` in my `composer.json`)
- the issue title is wrong, the error was thrown on **reading**, not on **saving**
- how about using `\DateTime::createFromInterface` instead of `\DateTime::createFromImmutable`?

**Comment by jacekkarczmarczyk** [comment]
Update: I've managed to install it with `composer require phpfastcache/phpfastcache:v9.x-dev`, I'm getting a different error though which now I'm investigating

**Comment by Geolim4** [comment]
I will fix them, thanks.

**Comment by Geolim4** [comment]
> how about using \DateTime::createFromInterface instead of \DateTime::createFromImmutable

Finally, I will keep the code as is, since I will made the same fix on the v8.1 and the v8.1 is still php 7.3 compatible, however `\DateTime::createFromInterface` has been added in PHP 8.

**Comment by jacekkarczmarczyk** [comment]
I've switched to DateTime as a workaround, when you release 9.1.1 I'll go back to DateTimeInterface, no hurry
Thanks!
